quaddtype/tests/test_quaddtype.py

```diff
@@ -1,33 +1,108 @@
+import pytest
+import sys
 import numpy as np
+import operator
 
-from quaddtype import QuadDType, QuadScalar
+from quaddtype import QuadPrecDType, QuadPrecision
 
 
-def test_dtype_creation():
-    assert str(QuadDType()) == "This is a quad (128-bit float) dtype."
+def test_create_scalar_simple():
+    assert isinstance(QuadPrecision("12.0"), QuadPrecision)
+    assert isinstance(QuadPrecision(1.63), QuadPrecision)
+    assert isinstance(QuadPrecision(1), QuadPrecision)
+
+
+def test_basic_equality():
+    assert QuadPrecision("12") == QuadPrecision(
+        "12.0") == QuadPrecision("12.00")
+
+
+@pytest.mark.parametrize("val", ["123532.543", "12893283.5"])
+def test_scalar_repr(val):
+    expected = f"QuadPrecision('{str(QuadPrecision(val))}')"
+    assert repr(QuadPrecision(val)) == expected
+
+
+@pytest.mark.parametrize("op", ["add", "sub", "mul", "truediv", "pow"])
+@pytest.mark.parametrize("other", ["3.0", "12.5", "100.0"])
+def test_binary_ops(op, other):
+    op_func = getattr(operator, op)
+    quad_a = QuadPrecision("12.5")
+    quad_b = QuadPrecision(other)
+    float_a = 12.5
+    float_b = float(other)
+
+    quad_result = op_func(quad_a, quad_b)
+    float_result = op_func(float_a, float_b)
 
+    assert np.abs(np.float64(quad_result) - float_result) < 1e-10
```

Comment:
I don't think this is testing much, let's chat more about how to test these ufuncs at our next meeting.

Reply:
We didn't get a chance to talk about this last night. The idea I had was to use sympy and/or mpfloat to calculate these results with *arbitrary* precision, then make sure the result you get is within the expected error bounds (1-4 ULP depending on the operation IIRC in the SLEEF docs).